File: ios/PDFToolkit/Models/RecentFile.swift
import Foundation

struct RecentFile: Identifiable, Codable {
    let id: String
    let name: String
    let fileUrlString: String
    let sizeBytes: Int64
    let timestamp: Double
    let toolType: ToolType
    let pageCount: Int

    var fileUrl: URL? {
        URL(string: fileUrlString)
    }

    var formattedSize: String {
        Self.formatBytes(sizeBytes)
    }

    static func formatBytes(_ bytes: Int64) -> String {
        if bytes <= 0 { return "0 B" }
        let units = ["B", "KB", "MB", "GB"]
        let digitGroups = Int(log10(Double(bytes)) / log10(1024.0))
        let unitIndex = max(0, min(digitGroups, units.count - 1))
        let value = Double(bytes) / pow(1024.0, Double(unitIndex))
        return String(format: "%.1f %@", value, units[unitIndex])
    }
}

File: android/app/src/main/java/com/paramdeep/pdftoolkit/data/model/RecentFile.kt
package com.paramdeep.pdftoolkit.data.model

data class RecentFile(
    val id: String,
    val name: String,
    val filePath: String,
    val sizeBytes: Long,
    val timestamp: Long,
    val toolType: ToolType,
    val pageCount: Int = 1
) {
    val formattedSize: String
        get() = formatBytes(sizeBytes)

    companion object {
        fun formatBytes(bytes: Long): String {
            if (bytes <= 0) return "0 B"
            val units = arrayOf("B", "KB", "MB", "GB")
            val digitGroups = (Math.log10(bytes.toDouble()) / Math.log10(1024.0)).toInt()
            val unitIndex = digitGroups.coerceIn(0, units.size - 1)
            val value = bytes / Math.pow(1024.0, unitIndex.toDouble())
            return String.format("%.1f %s", value, units[unitIndex])
        }
    }
}
